server/announcements.js

```diff
@@ -30,10 +29,12 @@ module.exports = function(imports) {
 
         util.audience.ensureIncludes(audience, req.user);
 
+        let content = util.normalizeDisplayedText(req.body.content);
```

Comment:
This does not work with the bold, italics, and other stuff provided by the announcement editor. In this case, all that is necessary is to do autolinker stuff, not prevent XSS, since the web client does that already.